resources/favEvents/favEvents.controller.js

```diff
@@ -1,6 +1,28 @@
 const FavEvent = require('./favEvents.model');
+const Event = require('../event/event.model');
 
 module.exports = {
+  async findFav(userId) {
+    const favEventsId = await FavEvent.find({ userId }, 'eventId userId', (err, docs) => {
+      if (err) {
+        return false;
+      }
+      return docs;
+    });
+    return favEventsId;
+  },
+  async findEvents(evArr) {
+    const result = await Promise.all(
+      evArr.map((event) => Event.findOne({
+        scrapedEventId: event.eventId,
+      }, (err, doc) => {
+        if (err) return false;
+        return doc._doc;
+      })),
+    )
+      .then((modelArr) => modelArr.map((model) => model._doc));
+    return result;
+  },
```

Comment:
@abidex4yemi @twumm @theallegrarr Do you think the efficiency of this findEvent controller will be optimal when the database becomes massive and the number of documents to be traversed increases?

Reply:
We would definitely refactor this in RC3, thanks for the feedback